### ISSUE TITLE
PgSubscriber reconnects when shutting down

### ISSUE THREAD

**Issue by zenios** [eclipse-vertx/vertx-sql-client#765]
When vertx is shutting down which will eventually shutdown the pool PgSubscribers with reconnect policy are not aware of it and try to reconnect resulting to exceptions being thrown

**Comment by vietj** [comment]
good point, likely PgSubscribers should not be called when vertx shutdown (likely using a close hooks). Note this seems valid for verticles undeploy
